### DISCUSSION TITLE
Passing multiple folders via command line

### DISCUSSION THREAD

**Discussion by alexanderpoppy** [dnGrep/dnGrep#990] [Q&A]
I am trying to pass two folders to dnGREP. I ran

```
"C:\Program Files\dnGrep\dnGREP.exe" -folder "D:\folder 1";"D:\folder 2"
```

but in dnGREP it looks like this 

![image](https://github.com/dnGrep/dnGrep/assets/50739239/37a4ea9f-b3db-4a88-aaf3-6c4bda5e6b0b)

and I get this when starting a search.

![image](https://github.com/dnGrep/dnGrep/assets/50739239/77394004-3c67-4499-9c98-06abae5ba26c)

Where is my mistake?

Version 4.0.151.0

**Comment by fade2gray** [comment]
What happens if you try enclosing the search parameter in another pair of double-quotes - `""D:\folder 1";"D:\folder 2""` ?

**Reply by alexanderpoppy** [reply]
"One or more arguments are not valid."

**Comment by doug24** [comment]
It looks like there is a bug in the command line processing.  If you paste `"D:\folder 1";"D:\folder 2"` into the Folder text box on the main window, it preserves the quotes in the correct positions and works. But the command line processor is having trouble figuring out where the -folder argument ends.

**Reply by doug24** [reply]
Moved this discussion to an issue.

**Comment by doug24** [comment]
This was another case of a breaking change between .NET Framework and .NET 7 that I missed. They changed the handling of command line arguments to parse them into an array to match how it is done in Linux.  But that parser stripped out the quotes on each path segment. There were unit tests for just this scenario of two quoted paths, but those tests assumed the command line arguments would be passed in exactly as the user entered them. 

I have changes in place to make a native Win32 call to get the unmodified command line arguments, and made some enhancements to the dnGrep command line parser so any of these formats will work:

`-folder "D:\folder 1";"D:\folder 2"`
`-folder ""D:\folder 1";"D:\folder 2""`
<s>`-folder D:\folder 1;D:\folder 2`</s>

Edit - had to remove the last, unquoted, one.  If a path contains a space, it must be quoted.  

These will work:
`-folder D:\folder1;"D:\folder 2"`
`-folder "D:\folder 1";D:\folder2`
